### PR TITLE
ProfileExpansion

### PR DESCRIPTION
**Pull request by Shromm-Gaind**
### As a user, I want to update my personal information so that my account details are always accurate.
**Issue: #14**

### LoginService Updates
- Added handling of `name` and `email` during user registration
- Updated `attemptLogin` to set full user details (including `name` and `email`) on successful login
- Added `updateUserDetails` to allow updating the user's `name` and `email`

### User Model Updates
- Added `id`, `name`, and `email` fields to `User` class
- Updated constructors to include new fields
- Added getters and setters for `name` and `email`
- Added email validation using `InternetAddress` for better format checks

### Unit Test Updates
- Expanded `UserTest` to include tests for `name` and `email`
- Added tests for constructor, getter, setter, and edge cases (null/empty fields)
- Added email validation tests

### CODE REVIEW

File: src/main/java/com/bookclub/controller/RegisterController.java [hunk 1]
```diff
@@ -25,11 +25,15 @@ public class RegisterController {
...
     private Label registerLabel;
     @FXML
+    private TextField usernameField;
+    @FXML
     private PasswordField passwordField;
     @FXML
     private PasswordField confirmPasswordField;
     @FXML
-    private TextField usernameField;
+    private TextField nameField;
```
Comment (jmccook96) [inline src/main/java/com/bookclub/controller/RegisterController.java:34]: Do we want to mandate name and email for registration? Open to the discussion (for the whole group), always thought of it as additional data that isn't strictly necessary- figured it would be content that could be added by the user in their account settings.

File: src/main/java/com/bookclub/service/LoginService.java [hunk 2]
```diff
@@ -92,4 +97,13 @@ public void dropCurrentUser() {
         currentUser = null;
         System.out.println("Current User dropped by LoginService.");
     }
+
+    public boolean updateUserDetails(String name, String email) {
```
Comment (jmccook96) [inline src/main/java/com/bookclub/service/LoginService.java:101]: The existence of updateUserDetails being a function of LoginService confuses me

Comment (Shromm-Gaind) [reply src/main/java/com/bookclub/service/LoginService.java:101]: That needs to be gone, that will be in a separate PR for account settings. Decided against it for this PR

File: src/main/java/com/bookclub/controller/RegisterController.java (hunk 1 above)
Comment (Shromm-Gaind) [reply src/main/java/com/bookclub/controller/RegisterController.java:34]: I had previously left it as optional. But my reasoning was that for a book club having a name and email would be kind of necessary.

File: src/main/java/com/bookclub/model/User.java [hunk 3]
```diff
@@ -2,100 +2,83 @@
...
-    /**
-     * Returns the password of the user.
-     *
-     * @return the password of the user
-     */
     public String getPassword() {
         return password;
     }
 
-    /**
```
Comment (jmccook96) [inline src/main/java/com/bookclub/model/User.java:88]: Please no delete documentation

File: src/main/java/com/bookclub/model/User.java [hunk 4]
```diff
@@ -2,100 +2,83 @@
...
+     * Name and email are initialized as empty strings.
      *
      * @param username the username of the user
      * @param password the password of the user
      * @throws IllegalArgumentException if the username or password is null or empty
      */
     public User(String username, String password) {
-        setUsername(username);
-        setPassword(password);
+        this(-1,username, password, "", "");
```
Comment (jmccook96) [inline src/main/java/com/bookclub/model/User.java:41]: Why?

File: src/main/java/com/bookclub/model/User.java [hunk 5]
```diff
@@ -2,100 +2,83 @@
...
+     * @param name     the name of the user
+     * @param email    the email of the user
      * @throws IllegalArgumentException if the username or password is null or empty
      */
-    public User(int id, String username, String password) {
+    public User(int id, String username, String password, String name, String email) {
         this.id = id;
         setUsername(username);
         setPassword(password);
+        this.name = name;
```
Comment (jmccook96) [inline src/main/java/com/bookclub/model/User.java:28]: Use setters in case we have enforcements for features of these fields please

File: src/main/java/com/bookclub/model/User.java [hunk 6]
```diff
@@ -2,100 +2,83 @@
...
 /**
  * The {@code User} class represents a user of the book club application.
- * Each user has a username and a password.
- * This class provides basic getter methods for accessing these attributes.
+ * Each user has a username, password, name, and email.
+ * This class provides getter and setter methods for accessing these attributes.
  */
 public class User {
-
-    // IF FIELDS GROW ENSURE USERDAO IS UPDATED TO MATCH.
```
Comment (jmccook96) [inline src/main/java/com/bookclub/model/User.java:10]: Retain this documenting comment please. Is an important reminder.

File: src/main/java/com/bookclub/service/LoginService.java [hunk 7]
```diff
@@ -68,12 +68,17 @@ public boolean attemptLogin(String username, String password) {
...
-    public boolean register(String username, String password) {
-        if (username == null || username.isEmpty() || password == null || password.isEmpty())
+    public boolean register(String username, String password, String name, String email) {
+        if (username == null || username.isEmpty() || password == null || password.isEmpty()) {
             return false;
+        }
 
-        User newUser = new User(username, password);
-        return !userAO.hasUser(newUser.getUsername()) && userAO.addUser(newUser);
+        User newUser = new User( -1,username, PasswordHasher.hashPassword(password), name, email);
```
Comment (jmccook96) [inline src/main/java/com/bookclub/service/LoginService.java:76]: We're misuing a constructor if we're intentionally providing invalid data on construction...

File: src/main/java/com/bookclub/mao/UserMAO.java [hunk 8]
```diff
@@ -7,55 +7,33 @@
 import java.util.Map;
 
 public class UserMAO implements IUserAO {
-    private final Map<Integer, User> users;
-    private int idIndex;
-    
+    private final Map<String, User> users;
```
Comment (jmccook96) [inline src/main/java/com/bookclub/mao/UserMAO.java:10]: You have once again damaged functionality I will have to return to make yours work...

File: src/main/java/com/bookclub/iao/IUserAO.java [hunk 9]
```diff
@@ -11,13 +11,6 @@ public interface IUserAO {
      */
     User findUserByUsername(String username);
 
-    /**
```
Comment (jmccook96) [inline src/main/java/com/bookclub/iao/IUserAO.java:14]: https://github.com/jmccook96/Organizer/pull/120/files?w=1#r1788529625

File: src/main/java/com/bookclub/dao/UserDAO.java [hunk 10]
```diff
@@ -5,107 +5,78 @@
...
-                    user.setId(newUserId);
+            stmt.setString(2, user.getPassword());
+            stmt.setString(3, user.getName());
+            stmt.setString(4, user.getEmail());
+
+            int affectedRows = stmt.executeUpdate();
+            if (affectedRows > 0) {
+                try (ResultSet generatedKeys = stmt.getGeneratedKeys()) {
+                    if (generatedKeys.next()) {
+                        user.setId(generatedKeys.getInt(1));
```
Comment (jmccook96) [inline src/main/java/com/bookclub/dao/UserDAO.java:44]: I don't understand the formatting choice of expanding affectedRows but collapsing newUserId 😆

File: src/main/java/com/bookclub/dao/UserDAO.java [hunk 11]
```diff
@@ -151,27 +121,16 @@ CREATE TABLE IF NOT EXISTS Users (
         }
     }
 
-    /**
-     * Attempts to find user from the created prepared statement.
-     * @param query The query to create a user from.
-     * @return Either the user found with the query, or null if failed.
-     * @throws SQLException if SQLException encountered.
-     */
-    private User findUserFromStatement(PreparedStatement query) throws SQLException {
```
Comment (jmccook96) [inline src/main/java/com/bookclub/dao/UserDAO.java:160]: Why cut this? Reduces duplicate code

File: src/main/java/com/bookclub/iao/IUserAO.java [hunk 12]
```diff
@@ -26,25 +19,18 @@ public interface IUserAO {
...
-     * Checks if a user with the passed id is contained.
-     * @param userId to find.
-     * @return if a user with the given id is found.
+     * Checks if a user is contained.
+     * @param user to find.
+     * @return If a user with the given username is found.
      */
-    boolean hasUser(int userId);
+    boolean hasUser(User user);
```
Comment (jmccook96) [inline src/main/java/com/bookclub/iao/IUserAO.java:33]: If I don't read the syntax highlight (I turn it off sometimes in my IDE), I'd assume hasUser is checking that the passed user entirely matches exactly with the version stored, not that we're pulling only the username to compare... If you want to do it this way as a wrapper add a function instead of changing existing behaviour

File: src/main/java/com/bookclub/controller/LoginController.java [hunk 13]
```diff
@@ -57,15 +57,22 @@ public void handleLogin(ActionEvent event) {
         String username = usernameField.getText();
         String password = passwordField.getText();
 
+        // Check for empty fields
+        if (username.isEmpty() || password.isEmpty()) {
+            showAlert("Login Failed", "Username and password cannot be empty.");
```
Comment (jmccook96) [inline src/main/java/com/bookclub/controller/LoginController.java:62]:
```suggestion
            showAlert("Login Failed", "Username or password cannot be empty.");
```

File: src/main/java/com/bookclub/controller/RegisterController.java [hunk 14]
```diff
@@ -57,32 +68,40 @@ private void initialize() {
...
 
+        // Check if passwords match
         if (!password.equals(confirmPassword)) {
             messageLabel.setText("Passwords do not match.");
             return;
         }
 
-        if (LoginService.getInstance().register(username, password)) {
+        // Validate email format
+        if (!email.isEmpty()) {
```
Comment (jmccook96) [inline src/main/java/com/bookclub/controller/RegisterController.java:91]: Not quite as optional here as suggested everywhere else

File: src/test/UserTest.java [hunk 15]
```diff
@@ -92,4 +144,14 @@ public void testConstructorInvalidPasswordNull() {
     public void testConstructorInvalidPasswordEmpty() {
         assertThrows(IllegalArgumentException.class, () -> new User("username", ""));
     }
+
+    @Test
+    public void testConstructorWithNameAndEmail() {
```
Comment (jmccook96) [inline src/test/UserTest.java:149]: This test is the same as the default we use on BeforeEach, making it redundant
